### DISCUSSION TITLE
Turns miniscope V4 led on and off alternately

### DISCUSSION THREAD

**Discussion by tudou2333333** [bonsai-rx/bonsai#1568] [Q&A]
I didn't find related topics in the posted discussions. The miniscope's recording needs to open led on the mices' brain and the long-time light could quench cells' fluorescence, but my experiment protocol has many intervals. 
So I'm trying to use two cycles to make miniscope turn on and turn off alternately. Here is my workflow. However, the miniscope can work at the first time I press the keydown input to begin and finish recording, then the led would not be turn on anymore. 
![image](https://github.com/bonsai-rx/bonsai/assets/102028902/70331695-5213-4287-87bc-f67c08c603f6)
![image](https://github.com/bonsai-rx/bonsai/assets/102028902/1d990fe7-a125-4d3a-8217-8dd0564f5ec1)
![image](https://github.com/bonsai-rx/bonsai/assets/102028902/d8817d5c-ff7f-4506-95bc-f3dcc712a680)
However, the miniscope can work at the first time I press the keydown input to begin and finish recording, then the led would not be turn on anymore. 
[miniscope_keydown.zip](https://github.com/bonsai-rx/bonsai/files/12734734/miniscope_keydown.zip)
Any help would be appreciated!

**Comment by ncguilbeault** [comment]
I think the issue here is that you are subscribing to 2 separate instances of the miniscope node, which can lead to the problems you described. I would suggest splitting your workflow into different parts - one that is accessing/saving the microscope data, another that is generating the start and stop sequence (SubscribeWhen, TakeUntil), and then finally Repeat. It would look something like this:

![main](https://github.com/bonsai-rx/bonsai/assets/26148508/480f6c9d-f714-4789-a825-5056889750df)
![MiniscopeAcquisition](https://github.com/bonsai-rx/bonsai/assets/26148508/48e5924f-2e49-4669-9c20-4f35c4e114d6)

**Reply by tudou2333333** [reply]
Thank you for your reply! 
The two V4 streams in my workflow have different led intensity, one is zero(To turn off miniscope), the other is with value(To turn on miniscope). And I need to record data only when the miniscope is turn on. If I split the workflow into different parts as you said, the led would not turn off which would produce phototoxicity to mice's cells.

**Reply by ncguilbeault** [reply]
There are a few ways to do it depending on what you need but one option is to add a TakeLast->"TurnOff" state following the TakeUntil node so that once data acquisition is finished, the state of the LED is reset and data acquisition stops.

[miniscope_acquisition_workflow.zip](https://github.com/bonsai-rx/bonsai/files/12740690/miniscope_acquisition_workflow.zip)

**Reply by tudou2333333** [reply]
It worked! Thank you so much!
